### PR TITLE
azurerm_local_network_gateway: support for `gateway_fqdn`

### PR DESCRIPTION
**Pull request by magodo**
## Test Result

```bash
💤 make testacc TEST=./azurerm/internal/services/network/tests TESTARGS='-run TestAccAzureRMLocalNetworkGateway_fqdn'    
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test ./azurerm/internal/services/network/tests -v -run TestAccAzureRMLocalNetworkGateway_fqdn -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccAzureRMLocalNetworkGateway_fqdn
=== PAUSE TestAccAzureRMLocalNetworkGateway_fqdn
=== CONT  TestAccAzureRMLocalNetworkGateway_fqdn
--- PASS: TestAccAzureRMLocalNetworkGateway_fqdn (127.35s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/tests       127.406s
```

### CODE REVIEW

**Comment by magodo** [comment]
Note that the CI error seems unrelated to this PR.

File: azurerm/internal/services/network/tests/local_network_gateway_resource_test.go [hunk 1]
```diff
@@ -236,6 +236,25 @@ func TestAccAzureRMLocalNetworkGateway_updateAddressSpace(t *testing.T) {
...
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMLocalNetworkGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMLocalNetworkGatewayConfig_fqdn(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLocalNetworkGatewayExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
```
Comment (tombuildsstuff) [inline azurerm/internal/services/network/tests/local_network_gateway_resource_test.go:253]: can we add an update step for this too?

**Review by tombuildsstuff — APPROVED** [review]
if we can add an update test this otherwise LGTM 👍

**Comment by magodo** [comment]
@tombuildsstuff Thank you for the feedback! I have updated the test per comment.

```bash
💤 make testacc TEST=./azurerm/internal/services/network/tests TESTARGS='-run TestAccAzureRMLocalNetworkGateway_fqdn'
==> Checking that code complies with gofmt requirements...  
==> Checking that Custom Timeouts are used...                                                                                                
==> Checking that acceptance test packages are used...                                                                                       
TF_ACC=1 go test ./azurerm/internal/services/network/tests -v -run TestAccAzureRMLocalNetworkGateway_fqdn -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccAzureRMLocalNetworkGateway_fqdn                 
=== PAUSE TestAccAzureRMLocalNetworkGateway_fqdn
=== CONT  TestAccAzureRMLocalNetworkGateway_fqdn                                                                                             
--- PASS: TestAccAzureRMLocalNetworkGateway_fqdn (197.12s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/tests       197.151s
```

File: azurerm/internal/services/network/tests/local_network_gateway_resource_test.go [hunk 2]
```diff
@@ -495,3 +521,24 @@ resource "azurerm_local_network_gateway" "test" {
...
 }
+
+func testAccAzureRMLocalNetworkGatewayConfig_fqdn(data acceptance.TestData, fqdn string) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-%d"
```
Comment (katbyte) [inline azurerm/internal/services/network/tests/local_network_gateway_resource_test.go:532]:
```suggestion
  name     = "acctestRG-network-%d"
```

File: website/docs/r/local_network_gateway.html.markdown [hunk 3]
```diff
@@ -40,14 +40,19 @@ The following arguments are supported:
...
-    connect.
-
 * `address_space` - (Required) The list of string CIDRs representing the
     address spaces the gateway exposes.
 
 * `bgp_settings` - (Optional) A `bgp_settings` block as defined below containing the
     Local Network Gateway's BGP speaker settings.
+    
+* `gateway_address` - (Optional) The IP address of the gateway to which to
+    connect.
```
Comment (katbyte) [inline website/docs/r/local_network_gateway.html.markdown:50]: grammer
```suggestion
* `gateway_address` - (Optional) The gateway IP address to connect with.
```

File: website/docs/r/local_network_gateway.html.markdown [hunk 4]
```diff
@@ -40,14 +40,19 @@ The following arguments are supported:
...
+    
+* `gateway_fqdn` - (Optional) The FQDN of the gateway to which to
+    connect.
```
Comment (katbyte) [inline website/docs/r/local_network_gateway.html.markdown:53]:
```suggestion
* `gateway_fqdn` - (Optional) The gateway FQDN to connect with.
```

**Review by katbyte — APPROVED** [review]
Thanks @magodo - aside from a couple minor comments this LGTM 👍

**Pull request merged**